### PR TITLE
Add --gui-script flag for running Python scripts with pythonw.exe on …

### PR DESCRIPTION
**Pull request by rajko-rad**
Addresses #6805

## Summary

This PR adds a `--gui-script` flag to `uv run` that allows running Python scripts with `pythonw.exe` on Windows, regardless of file extension. This solves the issue where users need to maintain duplicate `.py` and `.pyw` files to run the same script with and without a console window.

The implementation follows the pattern established by the existing `--script` flag, but uses `pythonw.exe` instead of `python.exe` on Windows. On non-Windows platforms, the flag is present but returns an error indicating it's Windows-only functionality.

Changes:
- Added `--gui-script` flag (Windows-only)
- Added Windows test to verify GUI script behavior
- Added non-Windows test to verify proper error message
- Updated CLI documentation

## Test Plan

The changes are tested through:

1. New Windows-specific test that verifies:
   - Script runs successfully with `pythonw.exe` when using `--gui-script`
   - Console output is suppressed in GUI mode but visible in regular mode
   - Same script can be run both ways without modification

2. New non-Windows test that verifies:
   - Appropriate error message when `--gui-script` is used on non-Windows platforms

3. Documentation updates to clearly indicate Windows-only functionality

### CODE REVIEW

**Comment by zanieb** [comment]
You can add an ignore for that Clippy complaint, if you want — or we will. We generally take our time about splitting those bools out into a struct.

File: crates/uv/tests/it/run.rs [hunk 1]
```diff
@@ -2737,6 +2737,79 @@ fn run_script_explicit_directory() -> Result<()> {
...
+            
+            # Close after 1 second
+            QTimer.singleShot(1000, app.quit)
+            sys.exit(app.exec_())
+        except Exception as e:
+            print(f"Error initializing GUI: {e}", file=sys.stderr)
+            sys.exit(1)
+    "#})?;
+
+    let output = context.run().arg("--script").arg("script").output()?;
```
Comment (zanieb) [inline crates/uv/tests/it/run.rs:2777]: Should this be using `--gui-script`?

**Comment by zanieb** [comment]
Nice! I made some minor tweaks and have one blocking question.

File: crates/uv/tests/it/run.rs (hunk 1 above)
Comment (rajko-rad) [reply crates/uv/tests/it/run.rs:2777]: you are right, and I'm slightly confused why/how we passed tests with that typo... Changing now and will see how it behaves!

**Comment by rajko-rad** [comment]
> Nice! I made some minor tweaks and have one blocking question.

Ok, blocking question addressed! For some reason I am getting a linting error that I think isn't real, see below?
Lint error:
image

Current codebase
image

There isn't a double space, not sure what is causing the error...

**Comment by zanieb** [comment]
The difference is some trailing whitespace

Screenshot 2024-11-27 at 10 00 39 AM

**Review by zanieb — APPROVED** [review]
Thank you!

**Comment by zanieb** [comment]
Just got super confused that this wasn't merged — not sure what happened there.

**Pull request merged**